src/test/py/bazel/bzlmod/repo_contents_cache_test.py

```diff
@@ -238,6 +240,41 @@ def testNoThrashingBetweenWorkspaces(self):
     _, _, stderr = self.RunBazel(['build', '@my_repo//:haha'], cwd=dir_a)
     self.assertNotIn('JUST FETCHED', '\n'.join(stderr))
 
+  def testGc(self):
+    self.ScratchFile(
+        'MODULE.bazel',
+        [
+            'repo = use_repo_rule("//:repo.bzl", "repo")',
+            'repo(name = "my_repo")',
+        ],
+    )
+    self.ScratchFile('BUILD.bazel')
+    self.ScratchFile(
+        'repo.bzl',
+        [
+            'def _repo_impl(rctx):',
+            '  rctx.file("BUILD", "filegroup(name=\'haha\')")',
+            '  print("JUST FETCHED")',
+            '  return rctx.repo_metadata(reproducible=True)',
+            'repo = repository_rule(_repo_impl)',
+        ],
+    )
+    # First fetch: not cached
+    _, _, stderr = self.RunBazel(['build', '@my_repo//:haha'])
+    self.assertIn('JUST FETCHED', '\n'.join(stderr))
+
+    # After expunging, but with a very quick GC delay & max age: cached
+    self.RunBazel(['clean', '--expunge'])
+    _, _, stderr = self.RunBazel(['build', '--repo_contents_cache_gc_max_age=0',
+                                  '--repo_contents_cache_gc_idle_delay=0',
+                                  '@my_repo//:haha'])
+    self.assertNotIn('JUST FETCHED', '\n'.join(stderr))
+
+    # GC'd while server is alive: not cached, but also no crash
+    time.sleep(0.5)
```

Comment:
This will definitely be flaky - I think it already is in CI on Windows. How about sleep looping until the repo dir is gone and only rerunning Bazel at that point?

Reply:
done. The Windows flake was actually because I forgot to copy the .bazelrc file over to the subdirectories in `testNoThrashingBetweenWorkspaces`, which caused the test to fail when run after some other tests (even without GC)!